src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php

```diff
@@ -92,6 +92,13 @@ public function load(array $configs, ContainerBuilder $container)
             $dir = dirname($reflection->getFileName()).'/Resources/views';
             if (is_dir($dir)) {
                 $this->addTwigPath($twigFilesystemLoaderDefinition, $dir, $bundle);
+
+                $classMock = new $class();
```

Comment:
Point is i think you need to traverse all possible parents, and register them for the current bundle `$dir`, so each parent works as expected.